processing/src/test/java/org/apache/druid/query/ordering/StringComparatorsTest.java

```diff
@@ -142,6 +142,16 @@ public void testNumericComparator()
     Assert.assertTrue(StringComparators.NUMERIC.compare("CAN'T PARSE THIS", "CAN'T TOUCH THIS") < 0);
   }
 
+  @Test
+  public void testVersionComparator()
```

Comment:
All of these tests also pass with `alphanumeric` sorting, maybe worth adding some test cases from the upstream project https://github.com/apache/maven/blob/a2b800de32cdb9adc1e64a43a0fc32e3ba878103/maven-artifact/src/test/java/org/apache/maven/artifact/versioning/DefaultArtifactVersionTest.java to illustrate the need for this comparator.

Reply:
Thanks, I'll add more test cases.